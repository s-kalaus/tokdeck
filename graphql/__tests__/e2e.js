process.env.NODE_ENV = 'test';
const gql = require('graphql-tag');
const { toPromise } = require('apollo-link');
const App = require('../../lib/app');
const { startTestServer } = require('./__utils');

const app = new App();

const LOGIN = gql`
  mutation customerLogin($login: String!, $password: String!) {
    customerLogin(login: $login, password: $password) {
      customer {
        customerId
      }
    }
  }
`;

describe('Graphql - e2e', () => {
  let stop;
  let graphql;

  beforeEach(async () => {
    const { server } = await app.graphqlStart();
    const testServer = await startTestServer(server);
    [stop, graphql] = [testServer.stop, testServer.graphql];
  });

  afterEach(() => stop());

  it('login', async () => {
    const res = await toPromise(
      graphql({
        query: LOGIN,
        variables: { login: 'sergey.kalaus@gmail.com', password: '12345678' },
      }),
    );
    expect(res).toMatchSnapshot();
  });
});
