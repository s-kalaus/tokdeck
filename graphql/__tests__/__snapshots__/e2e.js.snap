// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graphql - e2e login 1`] = `
Object {
  "data": Object {
    "customerLogin": Object {
      "customer": {
        "customerId": "1"
      },
    },
  },
}
`;
